state-chain/pallets/cf-broadcast/src/lib.rs

```diff
@@ -681,7 +684,36 @@ impl<T: Config<I>, I: 'static> Pallet<T, I> {
 			RequestFailureCallbacks::<T, I>::insert(broadcast_id, callback);
 		}
 
-		(broadcast_id, Self::threshold_sign(api_call, broadcast_id, true))
+		let _threshold_signature_id = Self::threshold_sign(api_call, broadcast_id, true);
+
+		broadcast_id
+	}
+
+	// TODO: make this into a governance extrinsic.
+	#[frame_support::transactional]
+	pub fn resign_aborted_broadcast(
+		broadcast_id: BroadcastId,
+	) -> Result<ThresholdSignatureRequestId, DispatchError> {
+		ensure!(
+			AbortedBroadcasts::<T, I>::mutate(|aborted: &mut Vec<u32>| {
+				// Convert to BTreeSet to remove duplicates.
+				let mut unique_ids = BTreeSet::<u32>::from_iter(aborted.iter().copied());
+				let removed = unique_ids.remove(&broadcast_id);
+				*aborted = Vec::from_iter(unique_ids.into_iter());
+				removed
+			}),
+			"Broadcast is not in the aborted list."
+		);
+		let (api_call, _) = ThresholdSignatureData::<T, I>::take(broadcast_id)
```

Comment:
```suggestion
		let (api_call, _) = ThresholdSignatureData::<T, I>::get(broadcast_id)
```
We will take on success in`clean_up_broadcast_storage`.

Reply:
Yes but it's inserted in the signature success callback - in a normal broadcast this doesn't exist until then.

Reply:
I have a refactor in mind that will make this clearer.